### PR TITLE
docs: Update adapter options in README

### LINKED ISSUE #241: Document how to configure push adapter with FCM
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server-push-adapter/security/policy).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server-push-adapter/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse Server Push Adapter](https://github.com/parse-community/parse-server-push-adapter/releases). 

### Issue Description

I had to look through code to figure out how to configure FCM.

Addressed by: https://github.com/parse-community/parse-server-push-adapter/pull/240

### PR DESCRIPTION
**Pull request by mman**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server-push-adapter/security/policy).
- [ ] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server-push-adapter/issues?q=is%3Aissue).

Closes #241

### Issue Description

This PR improves the README by adding examples how to configure push adapter for Apple, and Android devices. It perhaps should go to push guide as well, but I always try to lookup the config details here first.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,10 +57,79 @@ const parseServerOptions = {
...
+Delivering push notifications to Apple devices can be done either via Apple APNS (Apple Push Notification Service), or via FCM (Firebase Cloud Messaging) service. Note that each category of Apple devices require their own configuration section. Parse Server Push Adapter currently supports these types of Apple devices:
+
+- `ios` -> iPhone, iPad, and iPod touch apps
+- `osx` -> macOS, and macCatalyst apps
+- `tvos` -> tvOS apps
+
+
+Apple Push Notification Service requires a p8 encoded private key that can be generated and downloaded in Apple Developer Member Center under Certificates, Identifiers & Profiles.
+
+Google Firebase Cloud Messaging requires Firebase Service Account json encoded private key that can be downloaded in Firebase Console under your project Settings, and Cloud Messaging tab.
```
Comment (mtrezza) [inline README.md:85]:
```suggestion
Google Firebase Cloud Messaging requires a Firebase Service Account JSON encoded private key that can be downloaded in the Firebase Console under your project's Settings > Cloud Messaging tab.
```

File: README.md [hunk 2]
```diff
@@ -57,10 +57,79 @@ const parseServerOptions = {
...
+### Apple configuration
+
+Delivering push notifications to Apple devices can be done either via Apple APNS (Apple Push Notification Service), or via FCM (Firebase Cloud Messaging) service. Note that each category of Apple devices require their own configuration section. Parse Server Push Adapter currently supports these types of Apple devices:
+
+- `ios` -> iPhone, iPad, and iPod touch apps
+- `osx` -> macOS, and macCatalyst apps
+- `tvos` -> tvOS apps
+
+
+Apple Push Notification Service requires a p8 encoded private key that can be generated and downloaded in Apple Developer Member Center under Certificates, Identifiers & Profiles.
```
Comment (mtrezza) [inline README.md:83]:
```suggestion
Apple Push Notification Service requires a p8 encoded private key that can be generated and downloaded in the Apple Developer Center under Certificates > Identifiers & Profiles.
```

File: README.md [hunk 3]
```diff
@@ -57,10 +57,79 @@ const parseServerOptions = {
...
+Google Firebase Cloud Messaging requires Firebase Service Account json encoded private key that can be downloaded in Firebase Console under your project Settings, and Cloud Messaging tab.
+
+Here is an example configuration:
+
+```js
+const PushAdapter = require('@parse/push-adapter').default;
+const parseServerOptions = {
+  push: {
+    adapter: new PushAdapter({
+      android: {
```
Comment (mtrezza) [inline README.md:129]: Could you also add the current (soon decommissioned) way of configuring Android pushes? While it's still active and working, so developers have a reference.

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mman** [comment]
@mtrezza Let me know what you think.

File: README.md [hunk 4]
```diff
@@ -70,3 +70,96 @@ const parseServerOptions = {
...
+      }
+    })
+  },
+  /* Other Parse Server options */
+}
+```
+
+#### Migration from GCM to FCM (June 2024)
+
+Sending push notifications to Android devices was originally implemented using GCM (Google Cloud Messaging) API. GCM API was deprecated on June 20 2023 and will stop working after June 20 2024.
```
Comment (mtrezza) [inline README.md:142]: Is that correct? GCM has been decommissioned a while ago. I think we are migrating from FCM legacy API to the HTTP v1 API, which btw is also already deprecated, but I don't benefit a sunset date has been announced yet.

See https://firebase.google.com/docs/cloud-messaging/migrate-v1

We could actually add that link to the docs.

Comment (mman) [reply README.md:142]: Oh you may actually be right, I am honestly completely confused by the Google naming and was not involved in the Android push until recently... feel free to change as you understand it, I do not want to spread false info...

**Comment by mman** [comment]
Great changes @mtrezza, reads very cleanly now...

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.3.0](https://github.com/parse-community/parse-server-push-adapter/releases/tag/6.3.0)

**Pull request merged**